core/src/com/unciv/models/ModConstants.kt

```diff
@@ -11,25 +12,42 @@ class ModConstants {
-    val cityStrengthFromTechsFullMultiplier = 1.0
-    val cityStrengthFromGarrison = 0.2
+    var cityStrengthBase = 8.0
+    var cityStrengthPerPop = 0.4
+    var cityStrengthFromTechsMultiplier = 5.5
+    var cityStrengthFromTechsExponent = 2.8
+    var cityStrengthFromTechsFullMultiplier = 1.0
+    var cityStrengthFromGarrison = 0.2
  
     // Formula for Unit Supply:
     // Supply = unitSupplyBase (difficulties.json)
     //          unitSupplyPerCity * amountOfCities + (difficulties.json) 
     //          unitSupplyPerPopulation * amountOfPopulationInAllCities
     // unitSupplyBase and unitSupplyPerCity can be found in difficulties.json
     // unitSupplyBase, unitSupplyPerCity and unitSupplyPerPopulation can also be increased through uniques
-    val unitSupplyPerPopulation = 0.5
-    
+    var unitSupplyPerPopulation = 0.5
+
     // The minimal distance that must be between any two cities, not counting the tiles cities are on
     // The number is the amount of tiles between two cities, not counting the tiles the cities are on.
     // e.g. "C__C", where "C" is a tile with a city and "_" is a tile without a city, has a distance of 2.
     // First constant is for cities on the same landmass, the second is for cities on different continents.
-    val minimalCityDistance = 3
-    val minimalCityDistanceOnDifferentContinents = 2
-}
\ No newline at end of file
+    var minimalCityDistance = 3
+    var minimalCityDistanceOnDifferentContinents = 2
+
+    fun merge(other: ModConstants) {
+        if (other.maxXPfromBarbarians != defaults.maxXPfromBarbarians) maxXPfromBarbarians = other.maxXPfromBarbarians
+        if (other.cityStrengthBase != defaults.cityStrengthBase) cityStrengthBase = other.cityStrengthBase
+        if (other.cityStrengthPerPop != defaults.cityStrengthPerPop) cityStrengthPerPop = other.cityStrengthPerPop
+        if (other.cityStrengthFromTechsMultiplier != defaults.cityStrengthFromTechsMultiplier) cityStrengthFromTechsMultiplier = other.cityStrengthFromTechsMultiplier
+        if (other.cityStrengthFromTechsExponent != defaults.cityStrengthFromTechsExponent) cityStrengthFromTechsExponent = other.cityStrengthFromTechsExponent
+        if (other.cityStrengthFromTechsFullMultiplier != defaults.cityStrengthFromTechsFullMultiplier) cityStrengthFromTechsFullMultiplier = other.cityStrengthFromTechsFullMultiplier
+        if (other.cityStrengthFromGarrison != defaults.cityStrengthFromGarrison) cityStrengthFromGarrison = other.cityStrengthFromGarrison
+        if (other.unitSupplyPerPopulation != defaults.unitSupplyPerPopulation) unitSupplyPerPopulation = other.unitSupplyPerPopulation
+        if (other.minimalCityDistance != defaults.minimalCityDistance) minimalCityDistance = other.minimalCityDistance
+        if (other.minimalCityDistanceOnDifferentContinents != defaults.minimalCityDistanceOnDifferentContinents) minimalCityDistanceOnDifferentContinents = other.minimalCityDistanceOnDifferentContinents
+    }
```

Comment:
Is this really the best approach? The very-late end goal of the constants object, is to contain almost every single magic number in the entire game, so having to copy all of them over manually sounds like it will use an enormous amount of space.

Additionally, does it even make sense to allow non-baseruleset mods to change this? What if two non-baseruleset mods change the same constant, which one would the correct one be?

Reply:
Exactly! I had a variant making the class immutable, nice and allows a shorter merge with a helper function that also decides "which one would the correct one be" - but you'd still need a decision per member where I see no prepared shortcut short of reflection. The immutable variant was about 2 lines longer in the end so I ditched it. It involved using the `copy`offered by data classes to e.g. fix in the deprecated maxBarbXp. But you had one central place that could decide first, last or even average...

It all comes down to - if anyone simply publishes a mod that aims to change one of the values - and I can imagine changing that barbarians thingy to 0 or very large being popular - do we want them to work or not?

If not, we should warn!